src/rgw/driver/d4n/d4n_policy.cc

```
@@ -981,6 +967,11 @@ void LFUDAPolicy::cleaning(const DoutPrefixProvider* dpp)
 	    break;
 	  }//end-while (retry)
 	}
+         
+        if ((ret = cacheDriver->set_attr(dpp, head_oid_in_cache, RGW_CACHE_ATTR_DIRTY, "0", y)) < 0) {
+	  ldpp_dout(dpp, 0) << __func__ << "(): Failed to update dirty attr in cache, ret=" << op_ret << dendl;
+          // TODO: continue here?
```

@pritha-srivastava Should we `continue` in case of a failure here?

---

Shift set_attr to the place after policy->update() and ignore the error for now. follow the same order that you have done for data blocks - set_attr first and then directory entry